storage/innobase/srv/srv0srv.cc

```diff
@@ -2675,6 +2681,73 @@ srv_master_do_shutdown_tasks(
 	return(n_bytes_merged || n_tables_to_drop);
 }
 
+void
+srv_enable_temp_encryption_if_set()
+{
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+
+	if (is_shared_system_tablespace(srv_tmp_space.space_id())) {
+		/* there is no separate tablespace for temporary tables */
+		return;
+	}
+
+	if (srv_tmp_tablespace_encrypt) {
+
+		if (srv_read_only_mode) {
+			srv_tmp_tablespace_encrypt = false;
+			ib::error() << "Can't set temporary tablespace to be"
+				<< " encrypted in read-only mode.";
```

Comment:
InnoDB read only mode, to tell apart from the core server read only mode

Reply:
done